### PR TITLE
[TVMScript] IRBuilder methods for `PrimFunc`

### PR DESCRIPTION
**Pull request by cyx-6**
This PR introduces remaining IRBuilder methods for `PrimFunc`.

Co-authored-by: yongwww <yongcale@gmail.com>

### CODE REVIEW

File: python/tvm/script/ir_builder/tir/ir.py [hunk 1]
```diff
@@ -65,11 +358,121 @@ def evaluate(value: PrimExpr) -> None:
     return _ffi_api.Evaluate(value)  # pylint: disable=no-member # type: ignore
 
 
+def int8(expr: Optional[PrimExpr] = None) -> PrimExpr:
+    return _ffi_api.Int8(expr)  # pylint: disable=no-member # type: ignore
```
Comment (junrushao) [inline python/tvm/script/ir_builder/tir/ir.py:362]: please add some documentation here :-)

File: python/tvm/script/ir_builder/tir/ir.py [hunk 2]
```diff
@@ -65,11 +358,121 @@ def evaluate(value: PrimExpr) -> None:
...
+    return _ffi_api.UInt32(expr)  # pylint: disable=no-member # type: ignore
+
+
+def uint64(expr: Optional[PrimExpr] = None) -> PrimExpr:
+    return _ffi_api.UInt64(expr)  # pylint: disable=no-member # type: ignore
+
+
+def float8(expr: Optional[PrimExpr] = None) -> PrimExpr:
+    if not isinstance(expr, PrimExpr):
+        expr = convert(expr)
```
Comment (junrushao) [inline python/tvm/script/ir_builder/tir/ir.py:395]: why doesn't this logic appear in integer conversions above? can we simply remove it assuming the ffi takes care of conversion automatically?

File: src/script/ir_builder/tir/utils.h [hunk 3]
```diff
@@ -69,6 +69,14 @@ inline BlockFrame FindBlockFrame(const String& method) {
   throw;
 }
 
+inline tvm::tir::BufferRegion BufferRegionFromLoad(tvm::tir::BufferLoad buffer_load) {
```
Comment (junrushao) [inline src/script/ir_builder/tir/utils.h:72]: let's add some documentation

**Comment by junrushao** [comment]
Any updates?

File: python/tvm/script/ir_builder/tir/ir.py (hunk 1 above)
Comment (cyx-6) [reply python/tvm/script/ir_builder/tir/ir.py]: added

File: python/tvm/script/ir_builder/tir/ir.py (hunk 2 above)
Comment (cyx-6) [reply python/tvm/script/ir_builder/tir/ir.py]: deleted

File: src/script/ir_builder/tir/utils.h (hunk 3 above)
Comment (cyx-6) [reply src/script/ir_builder/tir/utils.h:96]: added

File: tests/python/unittest/test_tvmscript_ir_builder_tir.py [hunk 4]
```diff
@@ -28,18 +28,40 @@
...
+            T.func_attr({"key": "value"})
+            T.func_ret(tvm.ir.PrimType("int64"))
+            buffer_d = T.match_buffer(d, (64, 64), "int64")
+            T.preflattened_buffer(e, (32, 32), "int8", data=e.data)
             T.evaluate(0)
     # the prim_func generated by IRBuilder
     prim_func_actual = ib.get()
 
     # the expected prim_func
+    c_handle, c_buffer = tir.Var("c_handle", "handle"), tir.decl_buffer(
```
[the remark this answers is not in the document]
Comment (cyx-6) [reply tests/python/unittest/test_tvmscript_ir_builder_tir.py:65]: updated

**Comment by cyx-6** [comment]
> Any updates?

All have been updated.

**Review by junrushao — APPROVED** [review]
LGTM!

**Pull request merged**